src/overlays/actors/ovl_Demo_Gj/z_demo_gj.c

```diff
@@ -96,241 +85,1396 @@ static ColliderCylinderInitType1 sCylinderInit = {
+
+        theta += 0x2AAA;
+    }
+
+    DemoGj_PlayExplosionSfx(this, globalCtx);
+}
+
+s32 DemoGj_IsSceneInvalid(void) {
+    if (gSaveContext.sceneSetupIndex < 4) {
+        return 0; // return false; (?)
+    }
+    return 1; // return true; (?)
+}
+
+s32 DemoGj_FindGanon(DemoGj* this, GlobalContext* globalCtx) {
+    Actor* actor;
+
+    if (this->ganon == NULL) {
+        actor = globalCtx->actorCtx.actorLists[ACTORCAT_BOSS].head;
+
+        while (actor != NULL) {
+            if (actor->id == ACTOR_BOSS_GANON2) {
+                this->ganon = (BossGanon2*)actor;
+
+                // Demo_Gj_Search_Boss_Ganon %d: Discover Ganon !!!!
+                osSyncPrintf("Demo_Gj_Search_Boss_Ganon %d:ガノン発見!!!!\n", this->dyna.actor.params);
+                return true;
+            }
+            actor = actor->next;
+        }
+
+        // Demo_Gj_Search_Boss_Ganon %d: I couldn't find Ganon
+        osSyncPrintf("Demo_Gj_Search_Boss_Ganon %d:ガノン発見出来ず\n", this->dyna.actor.params);
+        return false;
+    }
+    //! @bug: Missing return value when `this->ganon` is already set.
+}
+
+void DemoGj_InitCommon(DemoGj* this, GlobalContext* globalCtx, CollisionHeader* header) {
+    GlobalContext* globalCtx2 = globalCtx;
+    Actor* actor = &this->dyna.actor;
+    DynaCollisionContext* colCtx_dyna = &globalCtx2->colCtx.dyna;
+    CollisionHeader* newHeader;
+
+    if (header != NULL) {
+        Actor_ProcessInitChain(actor, sInitChain);
+        DynaPolyActor_Init(&this->dyna, DPM_UNK);
+        newHeader = NULL;
+        CollisionHeader_GetVirtual(header, &newHeader);
+        this->dyna.bgId = DynaPoly_SetBgActor(globalCtx2, colCtx_dyna, actor, newHeader);
+    }
+}
+
+s32 DemoGj_InitSetIndexes(DemoGj* this, GlobalContext* globalCtx, s32 updateIndex, s32 drawIndex,
+                          CollisionHeader* header) {
+    if (!DemoGj_IsSceneInvalid()) {
+        this->updateIndex = updateIndex;
+        this->drawIndex = drawIndex;
+        DemoGj_InitCommon(this, globalCtx, header);
+        return 1; // return true; (?)
+    }
+    Actor_Kill(&this->dyna.actor);
+    return 0; // return false; (?)
+}
+
+void DemoGj_DrawCommon(DemoGj* this, GlobalContext* globalCtx, Gfx* displayList) {
+    if (kREG(0) == 0) {
+        GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+        OPEN_DISPS(gfxCtx, "../z_demo_gj.c", 1163);
+
+        func_80093D18(gfxCtx);
+
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(gfxCtx, "../z_demo_gj.c", 1165),
+                  G_MTX_PUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+        gSPDisplayList(POLY_OPA_DISP++, displayList);
+        gSPPopMatrix(POLY_OPA_DISP++, G_MTX_MODELVIEW);
+
+        CLOSE_DISPS(gfxCtx, "../z_demo_gj.c", 1169);
+    }
+}
+
+void func_8097911C(DemoGj* this, GlobalContext* globalCtx, Gfx* displayList) {
+    GlobalContext* globalCtx2 = globalCtx;
+    GraphicsContext* gfxCtx;
+
+    s16 x = this->rotationVec.x;
+    s16 y = this->rotationVec.y;
+    s16 z = this->rotationVec.z;
```

Comment:
Can this be turned into a Vec3s ?

Reply:
I tried with
```c
    vec.x = this->rotationVec.x;
    vec.y = this->rotationVec.y;
    vec.z = this->rotationVec.z;
```

but it mess up the stack.
![imagen](https://user-images.githubusercontent.com/7416381/106309757-56960280-6241-11eb-963b-ec6b4c6d3c52.png)
Notice the target code stores the data in decreasing order in the stack (`0x56`, `0x54`, and `0x52`), but the generated code goes in increasing order  (`0x50`, `0x52`, and `0x54`).

Do you know any better way?

I also tried with `vec = this->rotationVec;`, but it is way worse.